tensorflow/lite/micro/kernels/cmsis_nn/pooling.cc

```diff
@@ -15,99 +15,35 @@ limitations under the License.
+
   // Index to buffer for optimizations if applicable.
   int buffer_idx;
-
-  int32_t activation_min;
-  int32_t activation_max;
-  float activation_min_f32;
-  float activation_max_f32;
 };
 
-TfLiteStatus CalculateOpData(TfLiteContext* context,
-                             const TfLitePoolParams* params,
-                             const TfLiteTensor* input, TfLiteTensor* output,
-                             OpData* data) {
-  // input: batch, height, width, channel
-  int height = SizeOfDimension(input, 1);
-  int width = SizeOfDimension(input, 2);
-
-  int out_height, out_width;
-
-  data->padding = ComputePaddingHeightWidth(
-      params->stride_height, params->stride_width,
-      /*dilation_rate_height=*/1,
-      /*dilation_rate_width=*/1, height, width, params->filter_height,
-      params->filter_width, params->padding, &out_height, &out_width);
-
-  if (input->type == kTfLiteFloat32) {
-    CalculateActivationRange(params->activation, &data->activation_min_f32,
-                             &data->activation_max_f32);
-  } else {
-    TF_LITE_ENSURE_STATUS(CalculateActivationRangeQuantized(
-        context, params->activation, output, &data->activation_min,
-        &data->activation_max));
-    TFLITE_DCHECK_LE(data->activation_min, data->activation_max);
-  }
-
-  // Set buffer index to a reset value
-  data->buffer_idx = -1;
-
-  return kTfLiteOk;
-}
-
-void AverageEvalFloat(const TfLiteContext* context, const TfLiteNode* node,
-                      const TfLitePoolParams* params, const OpData& data,
-                      const TfLiteEvalTensor* input, TfLiteEvalTensor* output) {
-  float activation_min, activation_max;
-  CalculateActivationRange(params->activation, &activation_min,
-                           &activation_max);
-
-  PoolParams op_params;
-  op_params.stride_height = params->stride_height;
-  op_params.stride_width = params->stride_width;
-  op_params.filter_height = params->filter_height;
-  op_params.filter_width = params->filter_width;
-  op_params.padding_values.height = data.padding.height;
-  op_params.padding_values.width = data.padding.width;
-  op_params.float_activation_min = activation_min;
-  op_params.float_activation_max = activation_max;
-  reference_ops::AveragePool(op_params, tflite::micro::GetTensorShape(input),
-                             tflite::micro::GetTensorData<float>(input),
-                             tflite::micro::GetTensorShape(output),
-                             tflite::micro::GetTensorData<float>(output));
-}
-
 void AverageEvalQuantized(TfLiteContext* context, const TfLiteNode* node,
                           const TfLitePoolParams* params, const OpData& data,
                           const TfLiteEvalTensor* input,
                           TfLiteEvalTensor* output) {
   TFLITE_DCHECK(input->type == kTfLiteInt8);
 
-  RuntimeShape input_shape = tflite::micro::GetTensorShape(input);
+  RuntimeShape input_shape = micro::GetTensorShape(input);
```

Comment:
Somewhat unrelated to this change, do we plan to move away from RuntimeShape and MatchingDim for the sake of code size?

Reply:
I think we should, but we don't have a concrete plan for that right now.